TAV-619: First-time provisioning command

passcod opened the pull request:
### Changes

- Adds the concept of a "system user" which is used in those few instances we need a user but don't have a human one on hand
- Replaces the `setup` command with a full-featured `provision` command
- That takes care of creating users, facilities, users for facility sync auth, applying initial settings, and importing reference data
- Controlled by a single file documented here: https://beyond-essential.slab.com/posts/tamanu-provisioning-file-h1urgi86

### Checklist

- [x] Code is finished

_Upon merging:_

- [ ] Update the changelog (find it [here](https://github.com/beyondessential/tamanu/pulls?q=is%3Apr+is%3Aopen+label%3Arelease), or if the previous PR is merging into master then create a new release candidate PR for the next version following [the Slab doc](https://beyond-essential.slab.com/posts/merging-a-tamanu-ticket-okxp8s4s#h0y52-creating-a-release-candidate-pr))
  - [ ] with a ticket reference (e.g. `TAN-123: a one line description`, keep the lists sorted)
  - [ ] any config/settings changes and manual deployment steps
  - [ ] any db schema or other changes that could impact downstream data analysis

passcod committed:
Abstract loading a JSON/TOML/KDL file

passcod committed:
Replace setup with provisioning

passcod committed:
Replace initialUser with systemUser

passcod committed:
Add system user to existing installs

passcod committed:
Add graceful-quit option

mcccclean reviewed (changes requested):
This is really handy, replacing the `initialUser` setup with something more robust has been a long time coming.

Just have a few suggestions + noting there's a couple linter issues to get to

passcod committed:
Update packages/shared/src/migrations/1685403132663-systemUser.js

Co-authored-by: Tom McLean <tom.a.mclean@gmail.com>

passcod committed:
Add --provisioning option to serve/tasks

That way provisioning can be run immediately before starting a
server/taskrunner without having to call a separate process.

passcod committed:
systemUser() -> getSystemUser()

passcod committed:
UUID_NIL -> SYSTEM_USER_UUID

passcod committed:
Drop TOML support

passcod committed:
Fix tests/lints

passcod committed:
Remove obsolete snapshots
